androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/TextFields.kt

```diff
@@ -0,0 +1,46 @@
+package co.nimblehq.avishek.phong.kmmic.android.ui.screen.surveydetail
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import co.nimblehq.avishek.phong.kmmic.android.ui.common.PrimaryTextField
+import co.nimblehq.avishek.phong.kmmic.presentation.uimodel.SurveyAnswerUiModel
+
+@Composable
+fun TextFields(
+    surveyAnswerUiModels: List<SurveyAnswerUiModel>,
+    modifier: Modifier = Modifier
+) {
+    LazyColumn(
+        verticalArrangement = Arrangement.spacedBy(16.dp),
+        modifier = modifier
+    ) {
+        items(surveyAnswerUiModels.size) { index ->
+            var value by remember { mutableStateOf("") }
+
+            PrimaryTextField(
+                value = value,
+                onValueChange = {
+                    value = it
```

Comment:
Can you tell me how do you plan to retrieve the value of text fields from this? Because I don't see call backs in `TextFields` function.

Reply:
Implemented in the integrate PR #138. Please check this line: https://github.com/nimblehq/avishek-phong-kmm-ic/pull/138/files#diff-3aa24ff1fe9b7329064a9cca49f6da3a6dd933121af03389348f15e70791eb04R33